azurerm/resource_arm_dns_srv_record_test.go

```diff
@@ -99,42 +100,49 @@ func testCheckAzureRMDnsSrvRecordExists(name string) resource.TestCheckFunc {
 			return fmt.Errorf("Not found: %s", name)
 		}
 
-		conn := testAccProvider.Meta().(*ArmClient).rivieraClient
-
-		readRequest := conn.NewRequestForURI(rs.Primary.ID)
-		readRequest.Command = &dns.GetSRVRecordSet{}
+		srvName := rs.Primary.Attributes["name"]
+		zoneName := rs.Primary.Attributes["zone_name"]
+		resourceGroup, hasResourceGroup := rs.Primary.Attributes["resource_group_name"]
+		if !hasResourceGroup {
+			return fmt.Errorf("Bad: no resource group found in state for DNS SRV record: %s", srvName)
+		}
 
-		readResponse, err := readRequest.Execute()
+		conn := testAccProvider.Meta().(*ArmClient).dnsClient
+		resp, err := conn.Get(resourceGroup, zoneName, srvName, dns.SRV)
 		if err != nil {
-			return fmt.Errorf("Bad: GetSRVRecordSet: %s", err)
+			return fmt.Errorf("Bad: Get SRV RecordSet: %v", err)
 		}
-		if !readResponse.IsSuccessful() {
-			return fmt.Errorf("Bad: GetSRVRecordSet: %s", readResponse.Error)
+
+		if resp.StatusCode == http.StatusNotFound {
+			return fmt.Errorf("Bad: DNS SRV record %s (resource group: %s) does not exist", srvName, resourceGroup)
 		}
 
 		return nil
 	}
 }
 
 func testCheckAzureRMDnsSrvRecordDestroy(s *terraform.State) error {
-	conn := testAccProvider.Meta().(*ArmClient).rivieraClient
+	conn := testAccProvider.Meta().(*ArmClient).dnsClient
 
 	for _, rs := range s.RootModule().Resources {
 		if rs.Type != "azurerm_dns_srv_record" {
 			continue
 		}
 
-		readRequest := conn.NewRequestForURI(rs.Primary.ID)
-		readRequest.Command = &dns.GetSRVRecordSet{}
+		srvName := rs.Primary.Attributes["name"]
+		zoneName := rs.Primary.Attributes["zone_name"]
+		resourceGroup := rs.Primary.Attributes["resource_group_name"]
+
+		resp, err := conn.Get(resourceGroup, zoneName, srvName, dns.SRV)
 
-		readResponse, err := readRequest.Execute()
 		if err != nil {
-			return fmt.Errorf("Bad: GetSRVRecordSet: %s", err)
+			return nil
```

Comment:
I think we should be checking for the 404 above this, and then returning the error if it's present, rather than masking the fact there's an error here?